src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationBroadcaster.cs

```diff
@@ -102,7 +87,7 @@ public bool HasConnections
         {
             get
             {
-                return connectionManager != null && connectionManager.HasConnections;
+                return connectionManager != null && connectionManager.Connections.Count > 0;

```

Comment:
If I'm reading this correctly, connectionManager.Connections allocates and populates a new list just so that we can tell that connections exist.
TCPConnectionManager already has a bit more reasonable HasConnections property (which can be further implemented, see the comment about Count there). It's probably worth declaring it in INetworkConnectionManager (and then using it here).